src/app/views/panels/widgets/manager.tsx

```diff
@@ -403,6 +408,44 @@ export class WidgetManager implements Prototypes.Controls.WidgetManager {
       />
     );
   }
+  public inputFormat(
+    property: Prototypes.Controls.Property,
+    options: Prototypes.Controls.InputFormatOptions = {}
+  ) {
+    return (
+      <InputFormat
+        defaultValue={this.getPropertyValue(property) as string}
+        validate={(value) => {
+          if (value && value.trim() !== "") {
+            try {
+              getFormat()(value?.replace(/\{([^}]+)\}/g, "$1"));
```

Comment:
Can you write a common function for the regex - it also appears in [axis.ts](https://github.com/microsoft/charticulator/blob/ac7f4f9a608481feafda003b1bc6618561126ca6/src/core/prototypes/plot_segments/axis.ts#L176)